applications/tari_base_node/src/builder.rs

```diff
@@ -449,25 +444,21 @@ where
     let subscription_factory = Arc::new(subscription_factory);
     let comms_config = CommsConfig {
         node_identity: id.clone(),
-        peer_connection_listening_address: "/ip4/0.0.0.0/tcp/0".parse().expect("cannot fail"), /* TODO - make this
-                                                                                                * configurable */
-        socks_proxy_address: None,
-        control_service: ControlServiceConfig {
-            listening_address: id.public_address(),
-            socks_proxy_address: None,
-            public_peer_address: None, // TODO - make this configurable
-            requested_connection_timeout: Duration::from_millis(2000),
+        // TODO - make this configurable
+        transport_type: TransportType::Tcp {
+            listener_address: "/ip4/0.0.0.0/tcp/0".parse().expect("cannot fail"),
```

Comment:
```suggestion
            listener_address: id.public_address(),
```

Reply:
The public address is not always the same as the listener address. For instance, you may want to advertise a dns address/public ip and listen on '0.0.0.0:1234'. Listening on 0.0.0.0:0 is not correct either as you pointed out. 

More work is needed to make the transports configurable (especially because we'll be using tor). I think, for now, I should add `listener_address` configuration and plug that into there.

Reply:
Agreed, they should be two different settings